src/car/car.service.ts

```diff
@@ -93,21 +98,50 @@ export class CarService {
                         count: 1,
                     },
                 };
+                acc[`${monthName[month]},${year}`][car.car_id].percentage = ((acc[`${monthName[month]},${year}`][car.car_id].days / 30) * 100).toFixed(2);
             }
 
             return acc;
         }, {});
 
-        const sortedAllCarsUsage = Object.keys(allCarsUsage)
+        const sortedAllCarsUsage = Object.entries(allCarsUsage)
             .sort((a, b) => {
-                return monthName.indexOf(a) - monthName.indexOf(b);
+                const [monthA, yearA] = a[0].split(',');
+                const [monthB, yearB] = b[0].split(',');
+                if (yearA < yearB) {
+                    return -1;
+                } else if (yearA > yearB) {
+                    return 1;
+                } else {
+                    if (monthName.indexOf(monthA) < monthName.indexOf(monthB)) {
+                        return -1;
+                    } else if (monthName.indexOf(monthA) > monthName.indexOf(monthB)) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                }
```

Comment:
again every `else` is useless

Reply:
I apologize for using useless elses, however I have refactored it by using ternary operators.